apps/dotcom-worker/src/lib/routes/getRoomSnapshot.ts

```diff
@@ -1,15 +1,42 @@
 import { RoomSnapshot } from '@tldraw/tlsync'
 import { IRequest } from 'itty-router'
+import { getR2KeyForRoom } from '../r2'
 import { Environment } from '../types'
 import { createSupabaseClient, noSupabaseSorry } from '../utils/createSupabaseClient'
 import { fourOhFour } from '../utils/fourOhFour'
 import { getSnapshotsTable } from '../utils/getSnapshotsTable'
+import { R2Snapshot } from './createRoomSnapshot'
+
+function generateReponse(roomId: string, data: RoomSnapshot) {
+	return new Response(
+		JSON.stringify({
+			roomId,
+			records: data.documents.map((d) => d.state),
+			schema: data.schema,
+			error: false,
+		}),
+		{
+			headers: { 'content-type': 'application/json' },
+		}
+	)
+}
 
 // Returns a snapshot of the room at a given point in time
 export async function getRoomSnapshot(request: IRequest, env: Environment): Promise<Response> {
 	const roomId = request.params.roomId
 	if (!roomId) return fourOhFour()
 
+	// Get the room snapshot from R2
+	const snapshot = await env.ROOM_SNAPSHOTS.get(getR2KeyForRoom(roomId))
+
+	if (snapshot) {
+		const data = ((await snapshot.json()) as R2Snapshot)?.drawing as RoomSnapshot
+		if (data) {
+			return generateReponse(roomId, data)
+		}
```

Comment:
I guess it's also safe to return `fourOhFour` here. If we have a snapshot then it must be a new snapshot, so there's no need to check supabase.